tests/test_welopen.py

```diff
@@ -0,0 +1,269 @@
+import datetime
+
+import pandas as pd
+
+WELOPEN_CASES = [
+    (
+        """
+    DATES
+       1 MAY 2001 /
+    /
+
+    COMPDAT
+     'OP1' 33 110 31 31 'OPEN'  /
+    /
+
+    WELOPEN
+     'OP1' 'SHUT' /
+    /
+
+    TSTEP
+      1 /
+
+    COMPDAT
+     'OP1' 34 111 32 32 'OPEN' /
+    /
+
+    TSTEP
+      2 3 /
+
+    COMPDAT
+      'OP1' 35 111 33 33 'SHUT' /
+    /
+    """,
+        pd.DataFrame(
+            {
+                "WELL": {0: "OP1", 1: "OP1", 2: "OP1"},
+                "I": {0: 33, 1: 34, 2: 35},
+                "J": {0: 110, 1: 111, 2: 111},
+                "K1": {0: 31, 1: 32, 2: 33},
+                "K2": {0: 31, 1: 32, 2: 33},
+                "OP/SH": {0: "SHUT", 1: "OPEN", 2: "SHUT"},
+                "SATN": {0: 0, 1: 0, 2: 0},
+                "TRAN": {0: None, 1: None, 2: None},
```

Comment:
(could probably have added some `dropna` to the test-code to avoid listing these)

Reply:
fixed.